qualtran/bloqs/basic_gates/global_phase.py

```diff
@@ -46,7 +47,14 @@ def decompose_bloq(self) -> 'CompositeBloq':
         raise DecomposeTypeError(f"{self} is atomic")
 
     def adjoint(self) -> 'GlobalPhase':
-        return attrs.evolve(self, coefficient=complex(self.coefficient).conjugate())
+        from qualtran.resource_counting.symbolic_counting_utils import is_symbolic
+
+        coefficient = (
+            sympy.conjugate(self.coefficient)
+            if is_symbolic(self.coefficient)
+            else complex(self.coefficient).conjugate()
+        )
```

Comment:
worth adding as a helper function to `symbolic_counting_utils`?

Reply:
moved it there, and added a new type alias `SymbolicComplex`